src/pages/Swap/index.tsx

```diff
@@ -92,7 +92,40 @@ export function getIsValidSwapQuote(
   return !!swapInputError && !!trade && (tradeState === TradeState.VALID || tradeState === TradeState.SYNCING)
 }
 
-const formatAnalyticsEventProperties = (
+const formatApproveTokenTxnSubmittedEventProperties = (
+  approvalOptimizedTrade:
+    | Trade<Currency, Currency, TradeType>
+    | V2Trade<Currency, Currency, TradeType>
+    | V3Trade<Currency, Currency, TradeType>
+    | undefined
+) => {
+  if (!approvalOptimizedTrade) return {}
+  return {
+    chain_id: approvalOptimizedTrade.inputAmount.currency.chainId,
+    token_symbol: approvalOptimizedTrade.inputAmount.currency.symbol,
+    token_address: approvalOptimizedTrade.inputAmount.currency.isNative
+      ? NATIVE_CHAIN_ADDRESS
+      : approvalOptimizedTrade.inputAmount.currency.address,
+  }
+}
+
+const formatWrapTokenTxnSubmittedEventProperties = (
+  inputCurrency: Currency | null | undefined,
+  outputCurrency: Currency | null | undefined,
+  parsedAmount: CurrencyAmount<Currency> | undefined
+) => {
+  if (!inputCurrency || !outputCurrency || !parsedAmount) return {}
+  return {
+    token_in_address: inputCurrency.isNative ? NATIVE_CHAIN_ADDRESS : inputCurrency.address,
+    token_out_address: outputCurrency.isNative ? NATIVE_CHAIN_ADDRESS : outputCurrency.address,
+    token_in_symbol: inputCurrency.symbol,
+    token_out_symbol: outputCurrency.symbol,
+    chain_id: inputCurrency.chainId === outputCurrency.chainId ? inputCurrency.chainId : undefined,
```

Comment:
how could the `chain_ids` not be the same? feels like that should be handled before here

Reply:
i think fine to avoid this because you shoudn't need to update the swap code.

Reply:
true, removed check.